stdlib/public/core/FloatingPointTypes.swift.gyb

```diff
@@ -832,14 +816,29 @@ public prefix func - (x: ${Self}) -> ${Self} {
 // Construction from integers.
 @_transparent
 extension ${Self} {
-% for (srcBits, srcSigned) in allInts():
-%    That = intName(srcBits, srcSigned)
-%    ThatBuiltinName = builtinIntName(srcBits)
-%    sign = 's' if srcSigned else 'u'
+% for self_ty in all_integer_types(word_bits):
+%   That = self_ty.stdlib_name
+%   ThatBuiltinName = self_ty.builtin_name
+%   srcBits = self_ty.bits
+%   sign = 's' if self_ty.is_signed else 'u'
   public init(_ v: ${That}) {
     _value = Builtin.${sign}itofp_${ThatBuiltinName}_FPIEEE${bits}(v._value)
   }
-% end
+  
+%   if srcBits < SignificandBitCount:
+  @available(*, message: "Converting ${That} to ${Self} will always succeed.")
```

Comment:
You can use GYB to write tests and the usual filecheck mechanism to check the warning.  [For example](https://github.com/apple/swift/blob/master/test/stdlib/Runtime.swift.gyb).